src/arch/x86/tlb.hh

```diff
@@ -74,6 +74,12 @@ namespace X86ISA
 
         void setConfigAddress(uint32_t addr);
 
+        //concatenate Page Addr and pid
+        inline Addr concAddrPid(Addr vpn, uint64_t pid)
+        {
+          return (vpn | pid);
```

Comment:
Is VPN aligned to 4K so you know that the lower 12 bits are 0? If not, this isn't going to work. If so, then this is OK, but it won't support INVPCID very easily (which might be OK).

Reply:
vpn is aligned to page size, so at least the lower 12 bits should be 0 (assuming the smallest page size of 4K).